tests/unit/s2n_record_test.c

```diff
@@ -254,10 +254,11 @@ int main(int argc, char **argv)
 
         /* The last byte of out should indicate how much padding there was */
         uint8_t p = conn->out.blob.data[conn->out.write_cursor - 1];
-        EXPECT_EQUAL(5 + bytes_written + 20 + p + 1, s2n_stuffer_data_available(&conn->out));
+        const uint32_t remaining = 5 + bytes_written + 20 + p + 1;
+        EXPECT_EQUAL(remaining, s2n_stuffer_data_available(&conn->out));
 
         /* Check that the last 'p' bytes are all set to 'p' */
-        for (int j = 0; j <= p; j++) {
+        for (uint8_t j = 0; j <= p; j++) {
             EXPECT_EQUAL(conn->out.blob.data[5 + bytes_written + 20 + j], p);
         }
```

Comment:
This has the potential to loop forever if you use `uint8_t` (see https://godbolt.org/z/GzhvMaPWW)

It would be better to have j be `size_t` and then cast `(size_t) p` for the comparison.